### PR TITLE
docs(README): update mac installation

### LINKED ISSUE #22905: Problem with mac installation
There seems to be a problem with the brew install statement:
```
$ brew bundle install --file=etc/taskcluster/macos/Brewfile
Error: No Brewfile found
```

### PR DESCRIPTION
**Pull request by jsjoeio**
This clarifies the mac installation instructions. 

---

- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [x] These changes fix #22905 

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because it's a change to the `README`

### CODE REVIEW

**Comment by muodov** [comment]
Just bumped my head into this as well. Would be nice to merge this :)

File: README.md [hunk 1]
```diff
@@ -38,6 +38,16 @@ curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain none
...
+### Clone project
+
+```sh
+# using https
+git clone https://github.com/servo/servo.git
+
+# using ssh
+git@github.com:servo/servo.git
+```
+
```
Comment (paulrouget) [inline README.md:50]: I'm not against adding this but… this feels a bit unnecessary?

File: README.md [hunk 2]
```diff
@@ -56,7 +66,10 @@ Xcode version 10.2 or above is recommended.
 
 ##### On macOS (Homebrew)
 
+NOTE: run these steps after you've cloned the project locally.
+
 ``` sh
+cd servo
```
Comment (paulrouget) [inline README.md:72]: +1

File: README.md (hunk 1 above)
Comment (jsjoeio) [reply README.md:50]: Happy to remove! Your call

Comment (paulrouget) [reply README.md:50]: Yeah let's remove it :)

Thanks!

Comment (jsjoeio) [inline README.md:50]:
```suggestion
```
Remove

**Comment by jsjoeio** [comment]
Updated!

**Comment by paulrouget** [comment]
Thanks!

@bors-servo r+

**Pull request merged**